packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -497,8 +497,17 @@ export function TransferPanel() {
         } else {
           const amountRaw = utils.parseUnits(amount, 18)
 
+          if (isSmartContractWallet) {
+            showDelayedSCTxRequest()
+            // we can't call this inside the withdraw method because tx is executed in an external app
+            if (isFathomNetworkName(l2NetworkName)) {
+              trackEvent(`Deposit ETH to ${l2NetworkName} (Smart Contract)`)
+            }
+          }
+
           await latestEth.current.deposit({
             amount: amountRaw,
+            destinationAddress,
```

Comment:
Is this gonna work? Shouldn't we be using `depositTo`?

Reply:
This method calls `depositEth` in `token-bridge-sdk`. The logic for handling destinationAddress is implemented there.

But that made me realise that we should also check if `destinationAddress === walletAddress`. Right now we only check whether `destinationAddress` was provided, we should also make sure that it's different to `walletAddress` and only then use `depositTo`.

I will make that adjustment, one minute.